### ISSUE TITLE
"floating point exception" when trying to use 1200 resolution 

### ISSUE THREAD

**Issue by gregoiredx** [ThierryHFR/scangearmp2#65]
Hey, thanks a lot for this project which nicely improves the original scangear app!

It works well, except when I try to use the 1200 resolution, which I believe my scanner can handle (Canon Pixma TS6051).
![image](https://github.com/ThierryHFR/scangearmp2/assets/1934097/6d02a62c-d4e1-4bc8-85bb-bdd3a945c9e0)
The program crashes with `floating point exception (core dumped)`.

I tried to run with gdb and I get the following error:
```
Thread 1 "scangearmp2" received signal SIGFPE, Arithmetic exception.
0x000055555555af8b in CIJSC_start ()
```

Do you think this is a bug that could be fixed?

**Comment by ThierryHFR** [comment]
Sorry for the delay in replying.
I stand corrected. Can you tell me which distribution you use?

**Comment by gregoiredx** [comment]
Thanks for answering, nothing urgent. 
I'm on arch, I installed from the aur

**Comment by gregoiredx** [comment]
I see you made a fix, thanks! I'll try this later today

**Comment by gregoiredx** [comment]
Hello. Sorry for the delay, it took me longer than expected to retry this.
I tried again but I have a new problem now, it's failing with another message but I don't have any log, only this alert:
![image](https://github.com/user-attachments/assets/5c78c218-56ac-4e15-9592-5944dc1740ae)
I don't know how to get more details on the underlying problem.

Note: from MacOS it's possible to scan in 1200.

**Comment by gregoiredx** [comment]
Let me know if I can build/run the program with other options to get more details.

**Comment by ThierryHFR** [comment]
@gregoiredx 
The zoom factor for resolution 1200 is missing, added in the commit below : https://github.com/ThierryHFR/scangearmp2/commit/219de521feef669c126215849bb0471c0ad374fa.
Can you test, I don't have a Canon camera (at the moment) that uses scangearmp2?

**Comment by gregoiredx** [comment]
Sorry for the late response again.
I still have the same error :disappointed:  I'm starting to wonder if the lib can actually accept the 1200 resolution

**Comment by ThierryHFR** [comment]
Yes, the lib supports the 1200, I don't have a device to set it up. As soon as I get one, I'll fix it.
